jax/lax.py

```diff
@@ -992,6 +999,20 @@ def _add_transpose(t, x, y):
 ad.defbilinear_broadcasting(_brcast, mul_p, mul, mul)  # TODO
 
 
+def _safe_mul_translation_rule(c, x, y):
+  dtype = c.GetShape(x).numpy_dtype()
+  zero = c.Constant(onp.array(0, dtype=dtype))
+  out_shape = tuple(onp.maximum(c.GetShape(x).dimensions(),
+                                c.GetShape(y).dimensions()))
+  return c.Select(c.Or(c.Eq(x, zero), c.Eq(y, zero)),
+                  c.Broadcast(zero, out_shape),
+                  c.Mul(x, y))
+
+safe_mul_p = standard_binop([_num, _num], 'safe_mul',
+                            translation_rule=_safe_mul_translation_rule)
```

Comment:
Won't this need a JVP/transpose rule too?

Reply:
Actually the ad.defbilinear_broadcasting line after this one should handle that. But I’ll make sure it has the same rules set up as mul.

Reply:
I think by comparison to the definition of mul_p above we've got all the rules covered here. Thanks for checking!